### PR TITLE
MCR-2262 support requestHandler in solr uri

### PR DESCRIPTION
**Pull request by Possommi**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2262).

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 256aa422147eeed70581447183addb2812cc0671! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5878-pull_request-1043

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on b8298a16ff9b772764d925d858b7d72c3445f973! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5880-pull_request-1043

File: mycore-solr/src/main/java/org/mycore/solr/common/xml/MCRSolrQueryResolver.java [hunk 1]
```diff
@@ -18,46 +18,58 @@
...
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicReference;
+
 /**
- * <p>solr:{optional core}:query</p>
- * <p>solr:main:q=%2BobjectType%3Ajpjournal</p>
+ * <pre>
+ *  solr:{optional core}:query
+ *  solr:{optional core}:{optional requestHandler:&lt;requestHandler&gt;}:query
+ *  solr:q=%2BobjectType%3Ajpjournal
```
Comment (yagee-de) [inline mycore-solr/src/main/java/org/mycore/solr/common/xml/MCRSolrQueryResolver.java]: please provide an example for requestHandler

File: mycore-solr/src/main/java/org/mycore/solr/common/xml/MCRSolrQueryResolver.java [hunk 2]
```diff
@@ -18,46 +18,58 @@
...
 import org.mycore.common.content.MCRURLContent;
 import org.mycore.solr.MCRSolrClientFactory;
 import org.mycore.solr.search.MCRSolrURL;
 
+import javax.xml.transform.Source;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.URIResolver;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicReference;
+
```
Comment (yagee-de) [inline mycore-solr/src/main/java/org/mycore/solr/common/xml/MCRSolrQueryResolver.java]: please fix import order

File: mycore-solr/src/main/java/org/mycore/solr/search/MCRSolrURL.java [hunk 3]
```diff
@@ -18,8 +18,10 @@
 
 package org.mycore.solr.search;
 
-import static org.mycore.solr.MCRSolrConstants.SOLR_QUERY_PATH;
-import static org.mycore.solr.MCRSolrConstants.SOLR_QUERY_XML_PROTOCOL_VERSION;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.solr.client.solrj.SolrQuery.ORDER;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
```
Comment (yagee-de) [inline mycore-solr/src/main/java/org/mycore/solr/search/MCRSolrURL.java]: pleas fix import order

**Review by yagee-de — CHANGES_REQUESTED** [review]

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 153fb1152b4a6618761be41ffeec0f1f5c4dd55e! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5888-pull_request-1043

**Review by yagee-de — APPROVED** [review]
looks OK to me
Should we merge it to 2020.06.x?

**Comment by Possommi** [comment]
> Should we merge it to 2020.06.x?

not necessarily

**Pull request merged**